web/src/buddy_list.js

```diff
@@ -118,63 +131,89 @@ export class BuddyList extends BuddyListConf {
         $("#narrow-users-presence-container").toggleClass("no-display", subscribed_users_is_empty);
         $("#other-users-presence-container .buddy-list-subsection-header").toggleClass(
             "no-display",
-            subscribed_users.length === 0,
+            subscribed_users_is_empty,
         );
 
         const other_users_is_empty = this.$other_users_container.children().length === 0;
         $("#other-users-presence-container").toggleClass("no-display", other_users_is_empty);
         $("#narrow-users-presence-container .buddy-list-subsection-header").toggleClass(
             "no-display",
-            other_users.length === 0,
+            other_users_is_empty,
         );
 
-        // Invariant: more_keys.length >= items.length.
-        // (Usually they're the same, but occasionally keys
+        // Invariant: more_user_ids.length >= items.length.
+        // (Usually they're the same, but occasionally user_ids
         // won't return valid items.  Even though we don't
-        // actually render these keys, we still "count" them
+        // actually render these users, we still "count" them
         // as rendered.
 
-        this.render_count += more_keys.length;
+        this.render_count += more_user_ids.length;
         this.update_padding();
     }
 
     get_items() {
-        const $obj = this.$narrow_users_container.find(`${this.item_sel}`);
-        return $obj.map((_i, elem) => $(elem));
+        const $narrow_user_obj = this.$narrow_users_container.find(`${this.item_sel}`);
+        const $narrow_user_elems = $narrow_user_obj.map((_i, elem) => $(elem));
+
+        const $other_user_obj = this.$other_users_container.find(`${this.item_sel}`);
+        const $other_user_elems = $other_user_obj.map((_i, elem) => $(elem));
+
+        return [...$narrow_user_elems, ...$other_user_elems];
     }
 
     first_key() {
-        return this.keys[0];
+        if (this.narrow_user_ids.length) {
+            return this.narrow_user_ids[0];
+        }
+        if (this.other_user_ids.length) {
+            return this.other_user_ids[0];
+        }
+        return undefined;
     }
 
     prev_key(key) {
-        const i = this.keys.indexOf(key);
-
-        if (i <= 0) {
-            return undefined;
+        let i = this.narrow_user_ids.indexOf(key);
+        if (i > 0) {
+            return this.narrow_user_ids[i - 1];
         }
-
-        return this.keys[i - 1];
+        i = this.other_user_ids.indexOf(key);
+        if (i > 0) {
+            return this.other_user_ids[i - 1];
+        }
+        if (i === 0 && this.narrow_user_ids.length > 0) {
+            return this.narrow_user_ids.at(-1);
+        }
+        return undefined;
```

Comment:
This may have a bug, or maybe just these need better comments -- at least it looks like this logic has going to the "previous" element from the top of `narrow_user_ids` taking you to the bottom of the `narrow_user_ids` block as a wrap-around, and not to the bottom of `other_user_ids`? I'm not sure the intend but it seems inconsistent.

Reply:
@alya probably worth discussing if we want wrap-around to work here, since it's a decent amount of keyboard navigation code to re-review/test either way, and I'd like us to just do whatever version we want to end up with rather than doing it twice.

Reply:
Hmm, we don't wrap around in the left sidebar, so maybe not wrapping here is most consistent? But probably worth asking in #design, in case folks who use keyboard nav more have an opinion. (And we could change it in the left sidebar, of course.)

Reply:
I'd prefer to keep it as is for now, and switch to wrapping (in both places) if we want to as a separate task

Reply:
Yeah that seems mostly fine.